editor/src/core/es-modules/evaluator/evaluator.ts

```diff
@@ -18,38 +19,62 @@ function isEsModuleError(error: Error) {
   )
 }
 
+function transformToCommonJS(filePath: string, moduleCode: string): string {
+  const plugins = [BabelTransformCommonJS]
+  const result = Babel.transform(moduleCode, {
+    presets: ['es2015'],
+    plugins: plugins,
+    sourceType: 'module',
+    sourceFileName: filePath,
+  }).code
+  return result
+}
+
 function evaluateJs(
   filePath: string,
   moduleCode: string,
   partialModule: { exports: unknown },
   requireFn: (toImport: string) => unknown,
 ) {
-  // https://nodejs.org/api/modules.html#modules_module_exports
   let module = partialModule
-  let exports = module.exports
 
-  // https://nodejs.org/api/process.html#process_process_env
-  // This is a hacky solution, ideally we'd want a transpiler / loader that replaces process.env.NODE_ENV with a user-defined value
-  let process = {
-    env: {
-      NODE_ENV: 'production',
-    },
+  function firstErrorHandler(error: Error): void {
+    if (isEsModuleError(error)) {
+      const transpiledCode = transformToCommonJS(filePath, moduleCode)
```

Comment:
can Babel's `@babel/plugin-transform-modules-commonjs` run for a file that is already commonjs without destroying it? if so, we could pass every file through babel here.

that way, we could fix my `// This is a hacky solution, ideally we'd want a transpiler...` by also including this transformation: https://babeljs.io/docs/en/babel-plugin-transform-node-env-inline/

Reply:
I'm a little dubious of everything going through it given that the big blob `three` file takes 2-3 seconds to transpile. Which could _really_ stuff start up times.

Reply:
life in the browser is hard! :)